### PR TITLE
Minor DiagnosticSource http tests improvements

### PR DESCRIPTION
**Pull request by lmolkova**
Address review comments from #19863: improves comments and response dispose.
Also, redundant test is removed: same functionality is tested by other tests.

@stephentoub

### CODE REVIEW

File: src/System.Diagnostics.DiagnosticSource/tests/HttpHandlerDiagnosticListenerTests.cs [hunk 1]
```diff
@@ -49,16 +49,16 @@ public void TestReflectInitializationViaSubscription1()
                     client.GetAsync(Configuration.Http.RemoteEchoServer).Result.Dispose();
                 }
 
-                // Just make sure some events are written, to confirm we successfully subscribed to it. We should have 
-                // at least two events, one for request send, and one for response receive
+                // Just make sure some events are written, to confirm we successfully subscribed to it. 
+                // We should have exactly one Start and one Stop event
                 Assert.True(eventRecords.Records.Count == 2,
```
Comment (stephentoub) [inline src/System.Diagnostics.DiagnosticSource/tests/HttpHandlerDiagnosticListenerTests.cs]: Assert.Equal?

File: src/System.Diagnostics.DiagnosticSource/tests/HttpHandlerDiagnosticListenerTests.cs [hunk 2]
```diff
@@ -71,8 +71,8 @@ public void TestReflectInitializationViaSubscription2()
...
```
Comment (stephentoub) [inline src/System.Diagnostics.DiagnosticSource/tests/HttpHandlerDiagnosticListenerTests.cs]: Assert.Equal?

File: src/System.Diagnostics.DiagnosticSource/tests/HttpHandlerDiagnosticListenerTests.cs [hunk 3]
```diff
@@ -313,60 +301,11 @@ bool IsEnabled(string evnt, object arg1, object arg2)
...
-                    WebRequest request = ReadPublicProperty<WebRequest>(eventFields, "Request");
-                    Assert.Equal(request.GetType().Name, "HttpWebRequest");
-
-                    if (pair.Key == "System.Net.Http.Desktop.HttpRequestOut.Stop")
-                    {
-                        object response = ReadPublicProperty<WebResponse>(eventFields, "Response");
-                        Assert.Equal(response.GetType().Name, "HttpWebResponse");
-                    }
+                    (await client.GetAsync(Configuration.Http.SecureRemoteEchoServer)).Dispose();
+                    Assert.True(eventRecords.Records.Count > 0);
```
Comment (stephentoub) [inline src/System.Diagnostics.DiagnosticSource/tests/HttpHandlerDiagnosticListenerTests.cs]: Assert.InRange

**Comment by lmolkova** [comment]
@dotnet-bot test Innerloop netfx Windows_NT Debug x86 Build and Test please

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop netcoreapp Windows_NT Release x64 Build and Test please

**Pull request merged**